manifests/manage_app_pool.pp

```diff
@@ -16,15 +15,23 @@
   validate_re($start_mode, '^(OnDemand|AlwaysRunning)$')
   validate_bool($rapid_fail_protection)
 
+  if $apppoolrecycleperiodicminutes != undef {
+    validate_integer($apppoolrecycleperiodicminutes, 15372286728, 0) # powershell $([int64]::MaxValue) / 600000000, we're not dealing with negative
```

Comment:
You probably should make these numbers constants, variables or parameters, defined here or elsewhere